File: output/llama3.1-8b/avatar/Python/Java/atcoder_ABC133_B.java
import java.util.Scanner;
import java.lang.Math;

public class atcoder_ABC133_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");
        int n = Integer.parseInt(input[0]);
        int d = Integer.parseInt(input[1]);

        double[][] points = new double[n][d];
        for (int i = 0; i < n; i++) {
            input = scanner.nextLine().split(" ");
            for (int j = 0; j < d; j++) {
                points[i][j] = Double.parseDouble(input[j]);
            }
        }

        int count = 0;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                double dist = calculateDistance(points[i], points[j]);
                if (dist == Math.floor(dist)) {
                    count++;
                }
            }
        }

        System.out.println(count);
    }

    public static double calculateDistance(double[] point1, double[] point2) {
        double sum = 0;
        for (int i = 0; i < point1.length; i++) {
            sum += Math.pow(point1[i] - point2[i], 2);
        }
        return Math.sqrt(sum);
    }
}

File: output/llama3.1-8b/avatar/Java/Python/atcoder_ABC133_B.py
import math
import sys

def main():
    N, D = map(int, input().split())
    vectors = [[int(x) for x in input().split()] for _ in range(N)]

    answer = 0
    for i in range(N - 1):
        for j in range(i + 1, N):
            dist = sum((x - y) ** 2 for x, y in zip(vectors[i], vectors[j]))
            sq = math.sqrt(dist)
            if abs(sq - round(sq)) < 0.001:
                answer += 1

    print(answer)

if __name__ == "__main__":
    main()